src/main/java/com/trainigcenter/springtask/web/controller/MovieController.java

```diff
@@ -0,0 +1,155 @@
+package com.trainigcenter.springtask.web.controller;
+
+import com.trainigcenter.springtask.domain.Movie;
+import com.trainigcenter.springtask.service.MovieService;
+import com.trainigcenter.springtask.web.dto.Error;
+import com.trainigcenter.springtask.web.dto.MovieDto;
+import com.trainigcenter.springtask.web.exception.ForbiddenException;
+import com.trainigcenter.springtask.web.exception.NotFoundException;
+import com.trainigcenter.springtask.web.exception.WebException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
+
+    private static final Logger logger = LogManager.getLogger(MovieController.class);
+
+    private final MovieService movieService;
+    private final ModelMapper modelMapper;
+
+    @Autowired
+    public MovieController(MovieService movieService, ModelMapper modelMapper) {
+        this.movieService = movieService;
+        this.modelMapper = modelMapper;
+    }
+
+    @GetMapping
+    public List<MovieDto> getAll(@RequestParam(value = "page", defaultValue = "1") int page,
+                                 @RequestParam(value = "size", defaultValue = "2") int size,
+                                 @RequestParam(value = "rating", defaultValue = "8") int rating) {
+
+        if (page < 1 || size < 1) {
+            throw new NotFoundException("Page and size can't be less than 1");
+        }
+
+        return movieService.getAll(page, size)
+                           .stream()
+                           .map(this::convertToDto)
+                           .collect(Collectors.toList());
+    }
+
+    @GetMapping("/popular")
+    public List<MovieDto> getMostPopular(@RequestParam(value = "page", defaultValue = "1") int page,
```

Comment:
Basically, popularity and ordering by rating are the same things from API perspective. It would be really nice to combine `#getAll ()`, `#getMostPopular()` and `#getAllFilterByGenre()` as one API endpoint. But if you can’t do it, then let's leave it as it is.